src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -2184,7 +2184,7 @@ private void visitObjectType(
           false,
           Collections.<String>emptySet(),
           Collections.<String>emptySet(),
-          superClassFields);
+          Collections.<String>emptyList());
```

Comment:
This is the one place I'm not sure about.  The param here is "classTemplateTypeNames", so I don't know why we were passing in the superClassFields before.

Reply:
That's a bug, but likely we wanted `superClassFields` as 'skipNames', so we don't double emit inherited fields. That said I think we might have had this bug for long time, so at least add a comment.